README.md

```diff
@@ -1,31 +1,52 @@
 # Turnkey Rust SDK
 
-This repository contains tooling to interact with the Turnkey API using Rust.
+This repository contains tooling to interact with the Turnkey API using Rust, and is organized as a collection of Rust crates.
 
-Unlike other languages ([Typescript](https://github.com/tkhq/sdk), [Ruby](https://github.com/tkhq/ruby-sdk)), we do not yet offer a full SDK to make API requests to Turnkey using Rust.
-
-The main challenge when making requests to the Turnkey API is [request stamping](https://docs.turnkey.com/api-design/stamps). This repository includes code to sign API requests. Credits go to [@luca992](https://github.com/luca992) for the initial snippet!
-
-==> [`api_key_stamper`](./src/lib.rs)
+The two user-facing crates are:
+* [`client`](./client/): fully typed client to send requests to Turnkey
+* [`api_key_stamper`](./api_key_stamper/): used by `client` to stamp requests before they're sent to Turnkey
 
 ## Usage
 
 To make a request to Turnkey:
-* use any HTTP library ([`reqwest`](https://docs.rs/reqwest/latest/reqwest/) is a popular choice)
-* look at our [API reference](https://docs.turnkey.com/api-reference/overview) to learn about the JSON schema of our requests and responses
-* create a POST request body (all Turnkey API requests are POST requests). You can use [`serde_json`](https://docs.rs/serde_json/latest/serde_json/) to help with JSON serialization.
-* use the the provided `stamp` function to sign your request body with a `TurnkeyApiKey`
-* add the stamp value (bytes) inside of a new `X-Stamp` header, and attach it to the request
-* POST the request, read the response, and use the activity result
+* Load an API key:
+  ```rust
+  let api_key
+  ```
+* Create a new client:
+  ```rust
+  let client = tkhq_client::TurnkeyClient::new("https://api.turnkey.com", api_key, RetryConfig::default());
+  ```
+* Make a request (for example, a signature request)
+  ```rust
+  let signature_result = client.sign_raw_payload(
+    organization_id, // your organization ID
+    timestamp_ms, // time::SystemTime::now().duration_since(time::UNIX_EPOCH).unwrap().as_millis();
+    SignRawPayloadIntentV2 {
+        sign_with: address, // any Turnkey-generated address
+        payload: "hello from TKHQ".to_string(),
+        encoding: PayloadEncoding::TextUtf8,
+        hash_function: HashFunction::Keccak256,
+    },
+  ).await;
+  ```
 
 ## Examples
 
-For fully working examples, see our ['examples/' folder](./examples/).
+For fully working examples, see our ['examples/' folder](./examples/README.md).
+
+## Development
+
+This project uses `make` to encapsulate common tasks:
+* `make lint` will run rust fmt and clippy
+* `make test` runs the unit tests
```

Comment:
nit: A note about neeeding valid .env's for tests could be good

Reply:
tests do not require a valid env, but examples do. Good call!